Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1

```diff
@@ -281,47 +281,147 @@ function Set-TestAgentConfiguration
     {
-        Write-Verbose "Reboot required post test agent configuration, returning 3010" -Verbose
-        return 3010; # 3010 is exit code will indicate a reboot is required
+        Write-Verbose -Message "Trying to start TestAgent process interactively" -Verbose
+        InvokeDTAExecHostExe -Version $TestAgentVersion -MachineCredential $MachineUserCredential | Out-Null
     }
-    else
+
+    if (-not (IsDtaExecutionHostRunning))
     {
-        return $configOut.ExitCode
+        Write-Verbose -Message "DTAExecutionHost was not running interactively, marking the machine for reboot" -Verbose
+        return 3010
     }
+
+    return 0
+}
+
+function IsAnySessionActive()
+{
+    $wtssig = @'
+    namespace mystruct
+    {
+        using System;
+        using System.Runtime.InteropServices;
+
+        [StructLayout(LayoutKind.Sequential)]
+        public struct WTS_SESSION_INFO
+        {
+            public Int32 SessionID;
+
+            [MarshalAs(UnmanagedType.LPStr)]
+            public String pWinStationName;
+
+            public WTS_CONNECTSTATE_CLASS State;
+        }
+
+        public enum WTS_CONNECTSTATE_CLASS
+        {
+            WTSActive,
+            WTSConnected,
+            WTSConnectQuery,
+            WTSShadow,
+            WTSDisconnected,
+            WTSIdle,
+            WTSListen,
+            WTSReset,
+            WTSDown,
+            WTSInit
+        }
+    }
+'@
+
+    $wtsenumsig = @'
+        [DllImport("wtsapi32.dll", SetLastError = true)]
+        public static extern int WTSEnumerateSessions(
+            System.IntPtr hServer,
+            int Reserved,
+            int Version,
+            ref System.IntPtr ppSessionInfo,
+            ref int pCount);
+'@
+
+    $wtsopensig = @'
+        [DllImport("wtsapi32.dll", SetLastError = true)]
+        public static extern IntPtr WTSOpenServer(string pServerName);
+'@
+
+    $wtsSendMessagesig = @'
+        [DllImport("wtsapi32.dll", SetLastError = true)]
+        public static extern bool WTSSendMessage(
+            IntPtr hServer,
+            [MarshalAs(UnmanagedType.I4)] int SessionId,
+            String pTitle,
+            [MarshalAs(UnmanagedType.U4)] int TitleLength,
+            String pMessage,
+            [MarshalAs(UnmanagedType.U4)] int MessageLength,
+            [MarshalAs(UnmanagedType.U4)] int Style,
+            [MarshalAs(UnmanagedType.U4)] int Timeout,
+            [MarshalAs(UnmanagedType.U4)] out int pResponse,
+            bool bWait);
+'@
+
+    add-type  $wtssig
+    $wtsenum = add-type -MemberDefinition $wtsenumsig -Name PSWTSEnumerateSessions -Namespace GetLoggedOnUsers -PassThru
+    $wtsOpen = add-type -MemberDefinition $wtsopensig -name PSWTSOpenServer -Namespace GetLoggedOnUsers -PassThru
+    $wtsmessage = Add-Type -MemberDefinition $wtsSendMessagesig -name PSWTSSendMessage -Namespace GetLoggedOnUsers -PassThru
+
+    [long]$count = 0
+    [long]$ppSessionInfo = 0
+
+    $server = $wtsOpen::WTSOpenServer("localhost")
```

Comment:
and i would .... but it takes too much effort to get azure scenario working. http based standard env on azure is not supported